src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -369,133 +369,150 @@ internal unsafe void Start()
                     return;
                 }
 
-                ExecutionContext.Run(_context, delegate
+                ExecutionContext context = _context;
+                if (context is null)
                 {
+                    // Flow suppressed, just run here
+                    ProcessEvents(numEvents.ToInt32(), eventPaths, eventFlags, eventIds, watcher);
+                }
+                else
+                {
+                    ExecutionContext.Run(
+                        context,
+                        (object o) => ((RunningInstance)o).ProcessEvents(numEvents.ToInt32(), eventPaths, eventFlags, eventIds, watcher),
+                        this);
```

Comment:
It's not worth restarting CI for, but it's a little strange that we're trying to optimize for passing `this` via the object state, when we're still passing everything else in via the closure.